eva/binder/statement_binder.py

```diff
@@ -89,9 +91,15 @@ def _bind_create_index_statement(self, node: CreateIndexStatement):
 
         col_def = node.col_list[0]
         table_ref_obj = node.table_ref.table.table_obj
+
         col = [col for col in table_ref_obj.columns if col.name == col_def.name][0]
         assert col.type == ColumnType.NDARRAY, "Index input needs to be numpy array"
-        assert col.array_type == NdArrayType.FLOAT32, "Index input needs to be float32"
+
+        # Relax type checking for raw input table.
```

Comment:
Is this relaxing type checking for raw input table or udf case?

Reply:
For the UDF case. If the table already have feature vectors, their type needs to be float32. Otherwise, the type can be relaxed since it will go through the feature extractor, who outputs a float32 feature vector.

Reply:
The comment seems wrong then?

Reply:
In general, the index can be created on the types that are not float32 vectors. More accurately, it should be given the index method we are using, which is `HNSW`, we need either the column or the output of `node.udf_func` to be NDARRAY FLOAT32. For clarity, I think it will good to have this logic in the bind. 
```
If index_type == "HNSW":
   if node.udf_func is not None:
        # Check the output type is NDARRAY FLOAT32
   else:
        # Check the column type is NDARRAY FLOAT32
else:
   raise BinderError("Unsupported index method")
```
PS: I think we don't need the col type to be NDARRAY if there is `node.udf_func`.

Reply:
Got it. FAISS itself also has other index types besides `HNSW`. The FLOAT32 type checking applies to those other index types as well, so I think the `if index_type == "HNSW":` is probably not needed?

Reply:
I think that the query `CREATE INDEX ON table (ColorDetector(col));` should be legit. It's okay we are not supporting it now with FAISS. But the binder should raise `unsupported index` instead of `Index input needs to be numpy array FLOAT32` for clarity.